symengine/matrix.h

```diff
@@ -172,6 +172,12 @@ class DenseMatrix : public MatrixBase
                                 unsigned row_start, unsigned col_start,
                                 unsigned row_end, unsigned col_end,
                                 unsigned row_step, unsigned col_step);
+    friend void row_join(const DenseMatrix &A, const DenseMatrix &B,
+                         DenseMatrix &C);
+    friend void col_join(const DenseMatrix &A, const DenseMatrix &B,
+                         DenseMatrix &C);
+    friend void row_del(DenseMatrix &A, unsigned k);
+    friend void col_del(DenseMatrix &A, unsigned k);
```

Comment:
These shouldn't be `friend` functions. Also, make all 4 inplace functions. First 2 functions has an output `C` and the last 2 modifies the input inplace